File: test-suite/src/test/java/io/micronaut/docs/function/aws/test/SampleRequestHandler.java
package io.micronaut.docs.function.aws.test;

//tag::clazz[]
import io.micronaut.context.ApplicationContext;
import io.micronaut.function.aws.MicronautRequestHandler;

public class SampleRequestHandler extends MicronautRequestHandler<String, String> {
    // Used in AWS
    public SampleRequestHandler() {
    }

    // Used in tests
    public SampleRequestHandler(ApplicationContext applicationContext) {
        super(applicationContext);
    }

    @Override
    public String execute(String input) {
        return "Hello " + input;
    }
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/function/aws/test/SampleRequestHandler.kt
package io.micronaut.docs.function.aws.test

//tag::clazz[]
import io.micronaut.context.ApplicationContext
import io.micronaut.function.aws.MicronautRequestHandler

class SampleRequestHandler : MicronautRequestHandler<String, String> {
    // Used in AWS
    constructor() : super()

    // Used in tests
    constructor(applicationContext: ApplicationContext) : super(applicationContext)

    override fun execute(input: String): String {
        return "Hello $input"
    }
}
//end::clazz[]
